addons/resource/models/resource_calendar.py

```diff
@@ -589,6 +591,10 @@ def _get_unusual_days(self, start_dt, end_dt, company_id=False):
         works = {d[0].date() for d in self._work_intervals_batch(start_dt, end_dt, domain=domain)[False]}
         return {fields.Date.to_string(day.date()): (day.date() not in works) for day in rrule(DAILY, start_dt, until=end_dt)}
 
+    def _is_fully_flexible_working_hours(self):
+        """ employee has a fully flexible schedule if flagged flexible_hours and has hours_per_day set to 0.0 """
+        return self.flexible_hours and not self.hours_per_day
```

Comment:
either make this in batch, or put ensure_one..  in batch is preferred

Reply:
Given that the function is called on each individual calendar in the planning module, I opted for adding `ensure_one`. 
[here](https://github.com/odoo-dev/enterprise/blob/5cba18161b789e6222158a151e8a5e1eb2f2d265/planning/models/planning.py#L274) and [here](https://github.com/odoo-dev/enterprise/blob/5cba18161b789e6222158a151e8a5e1eb2f2d265/planning/models/planning.py#L2236) in enterprise.